### PR TITLE
fix: `label_values` w/ empty filters in prom datasource variable query

### LINKED ISSUE #78186: An error shows when you have a data source selected but no label filters
The prometheus template variables query label_values(<filters>, __name__) shows an error when there are not label filters passed to it. Not sure what the best way is to fix it. Either we fix it in Prometheus or we add some new logic that only adds this variable to the MetricSelectScene after we have at least one filter.

### PR DESCRIPTION
**Pull request by darrenjaneczek**
**What is this feature?**

When constructing queries for variable queries using scenes, and you want to use something similar to an adhoc filter for the `label_values`, there is a chance that the ad hoc filter might be empty, being represented by `{}`.

This fix allows the prometheus datasource to handle empty filters by behaving as if no filters have been chosen instead of encountering a backend query error.

**Why do we need this feature?**

This allows data trails to provide label names before an explicit filter has been set.

**Who is this feature for?**

Users of data trails.

**Which issue(s) does this PR fix?**:

- Fixes #78186

BEFORE:
image

AFTER:
image

This approach won't solve all cases and is currently prioritized to work with scenes, as is done through the data-trails prototype. Currently, if you attempt to use an ad-hoc filter in a normal dashboard, the interpolated query on this line:

See:
- https://github.com/grafana/grafana/blob/a3827333514de8c198f2ac842ecdf3e4d5c6418c/public/app/plugins/datasource/prometheus/variables.ts#L52
will still end up with `label_values(,__name__)` for `label_values(${filter},__name__}` which will not be detected by the enhancements to `PrometheusMetricFindQuery` in this PR. Further, even if the ad hoc filter does have selections, they are not currently being represented in the interpolation of `${filter}`. So more work will be required for this to work generally in dashboards.

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
  - Start Grafana via `GF_FEATURE_TOGGLES_ENABLE=datatrails make run`
  - Before this PR, `http://localhost:3000/data-trails/trail?var-ds=gdev-prometheus` would result in an empty trail due to a query error (empty filter
  - With this PR, the trail is populated with a set of initial labels
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.
  - [ ] I will need some feedback to evaluate if we want to update the docs and make a note
- [ ] Are there other spots where we might want to ensure similar behavior is set up?

### CODE REVIEW

File: public/app/plugins/datasource/prometheus/metric_find_query.ts [hunk 1]
```diff
@@ -48,7 +48,8 @@ export default class PrometheusMetricFindQuery {
 
     const labelValuesQuery = this.query.match(labelValuesRegex);
     if (labelValuesQuery) {
-      if (labelValuesQuery[1]) {
+      const hasEmptyFilter = () => labelValuesQuery[1].split(' ').join('') === '{}';
```
Comment (itsmylife) [inline public/app/plugins/datasource/prometheus/metric_find_query.ts:51]: I think this should be outside of this method, ideally a helper function outside of the class.

**Comment by darrenjaneczek** [comment]
To elaborate on a point above:

>See:
>- https://github.com/grafana/grafana/blob/a3827333514de8c198f2ac842ecdf3e4d5c6418c/public/app/plugins/datasource/prometheus/variables.ts#L52
will still end up with `label_values(,__name__)` for `label_values(${filter},__name__}` which will not be detected by the enhancements to `PrometheusMetricFindQuery` in this PR. Further, even if the ad hoc filter does have selections, they are not currently being represented in the interpolation of `${filter}`. So more work will be required for this to work generally in dashboards.

File: public/app/plugins/datasource/prometheus/metric_find_query.ts (hunk 1 above)
Comment (darrenjaneczek) [reply public/app/plugins/datasource/prometheus/metric_find_query.ts]: I moved it out, and made a few things more descriptive.

Comment (darrenjaneczek) [reply public/app/plugins/datasource/prometheus/metric_find_query.ts]: I also ensured that the tests perform the various tests cases in the array it iterates (vs. just using that to make test descriptions)
- https://github.com/grafana/grafana/pull/78242/commits/4861ad7a29de94a0a289a734ce92fa14b78ff11a

**Review by bohandley — APPROVED** [review]
LGTM!

**Pull request merged**